src/fields/addCounter.jsx

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react';
+
+export default function addCounter(TextInput) {
+
+  return class Wrapper extends Component {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        charCount: 0,
+        wordCount: 1,
+        value: ''
+      }
+
+      this.textInputProps = Object.assign({}, this.props);
```

Comment:
We should not need to do this here. We should do this in `render` when we bootstrap the wrapped component. After all, the props can change during the lifetime of this component, and putting this filter here will *only* kick in the first time the component is built, rather than every time it renders.